go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/controller-runtime
 
-go 1.13
+go 1.15
```

Comment:
The presubmits still run with go 1.13, so we should probably change that first: https://github.com/kubernetes/test-infra/blob/f6e9dbfd16c179a89c84f98481326884ca69bcc4/config/jobs/kubernetes-sigs/controller-runtime/controller-runtime-presubmits-master.yaml#L12

and why does specifying a newer go version than used in the tests not make the tests fail?

Reply:
From https://golang.org/doc/go1.12#modules:

> The go directive in a go.mod file now indicates the version of the language used by the files within that module. It will be set to the current release (go 1.12) if no existing version is present. If the go directive for a module specifies a version newer than the toolchain in use, the go command will attempt to build the packages regardless, and will note the mismatch only if that build fails.

Reply:
I'll open a PR in test-infra to update the version

Reply:
https://github.com/kubernetes/test-infra/pull/19383